src/App.tsx

```diff
@@ -1,95 +1,144 @@
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+export class App extends React.Component<{}, State> {
+  state = {
+    sortType: SortType.NONE,
+    isReversed: false,
+  };
 
-      <ul>
-        <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
-        </ul>
-      </ul>
-    </div>
-  );
-};
+  sortByAlphabetically = () => {
+    this.setState({ sortType: SortType.ALPHABET });
+  };
+
+  sortByLenth = () => {
+    this.setState({ sortType: SortType.LENGTH });
+  };
+
+  reset = () => {
+    this.setState({
+      sortType: SortType.NONE,
+      isReversed: false,
+    });
+  };
+
+  reverse = () => {
+    this.setState((state) => ({ isReversed: !state.isReversed }));
+  };
+
+  render() {
+    const {
+      sortType,
+      isReversed,
+    } = this.state;
+
+    const preparedList = getReorderedGoods(
+      goodsFromServer,
+      { sortType, isReversed },
+    );
+
+    return (
+      <div className="section content">
+        <div className="buttons">
+          <button
+            type="button"
+            className={classNames(
+              'button is-info',
+              { 'is-light': sortType !== SortType.ALPHABET },
+            )}
+            onClick={this.sortByAlphabetically}
+          >
+            Sort alphabetically
+          </button>
+
+          <button
+            type="button"
+            className={classNames(
+              'button is-success',
+              { 'is-light': sortType !== SortType.LENGTH },
+            )}
+            onClick={this.sortByLenth}
+          >
+            Sort by length
+          </button>
+
+          <button
+            type="button"
+            className={classNames(
+              'button is-warning',
+              { 'is-light': !isReversed },
+            )}
+            onClick={this.reverse}
+          >
+            Reverse
+          </button>
+
+          {(sortType !== SortType.NONE || isReversed)
+            ? (
+              <button
+                type="button"
+                className={classNames('button is-info',
+                  { 'is-light': sortType !== SortType.NONE })}
+                onClick={this.reset}
+              >
+                Reset
+              </button>
+            )
+            : '' }
```

Comment:
```suggestion
          {(sortType !== SortType.NONE || isReversed) && (
            <button
              type="button"
              className={classNames('button is-info',
                { 'is-light': sortType !== SortType.NONE })}
              onClick={this.reset}
            >
              Reset
            </button>
          )}
```

in such case, better to use conditional rendering instead of ternary